### DISCUSSION TITLE
Potential implementation issue.

### DISCUSSION THREAD

**Discussion by comfyanonymous** [TheMistoAI/MistoControlNet-Flux-dev#24] [Announcements]
https://github.com/TheMistoAI/MistoControlNet-Flux-dev/blob/main/modules/utils.py#L372

Shouldn't this be:

```
img = img + block_controlnet_hidden_states[index_block % 3]
```

**Comment by baizh0u** [comment]
Could it lead to better result?

**Comment by bonlime** [comment]
While current implementation seems to be wrong based on code, for some reason "fixing" it using this:
`img = img + block_controlnet_hidden_states[index_block % 3]`
only makes the results worse, which is very strange
